.github/workflows/misc.yml

```
@@ -8,6 +8,33 @@ jobs:
     - uses: actions/checkout@v4
     - name: Check for currency.NewPair(BTC, USD) used instead of currency.NewBTCUSD
       run: |
-        grep -r -n -E "currency.NewPair\(currency.BTC, currency.USDT?\)" * || exit 0
+        grep -r -n --color=always -E "currency.NewPair\(currency.BTC, currency.USDT?\)" * || exit 0
         echo "::error::Replace currency.NewPair(BTC, USD*) with currency.NewBTCUSD*()"
         exit 1
+
+    - name: Check for missing postfix `f` format func variant for testify assertions
+      run: |
+        grep -r -n -P --include="*.go" --color=always '(assert|require)\.[A-Za-z_]\w*?(?<!f)\((?:(?!fmt\.Sprintf).)*%.*' || exit 0
+        echo "::error::Replace func with the `…f` func variant (e.g. Equalf, Errorf)"
+        exit 1
+
+    - name: Check for quoted and backticked %s usage in format specifier strings
+      run: |
+        grep -r -n --include='*.go' --color=always -E "[\`']%s[\`']" . || exit 0
+        echo "::error::Replace '%s' or `%s` format specifier with %q"
+        exit 1
+
+    - name: Check for testify `require… "should"` and `assert… "must"` message consistency
+      run: |
+        exit_code=0
+    
+        echo "Checking for 'should' in require messages..."
+        grep -r -n --include="*.go" --color=always -E 'require\.[A-Za-z0-9_]+.*"[^"]*should[^"]*"' . && exit_code=1 || true
+ 
+        echo "Checking for 'must' in assert messages..."
+        grep -r -n --include="*.go" --color=always -E 'assert\.[A-Za-z0-9_]+.*"[^"]*must[^"]*"' . && exit_code=1 || true
+
+        if [ $exit_code -eq 1 ]; then
+          echo "::error::Replace \"should\" in require messages and \"must\" in assert messages"
+          exit 1
+        fi
```

🥃  Actually, whilst we're here, could haz?
```suggestion
        fi

    - name: Check for error.Is(err, nil) in tests
      run: |
        grep -r -n --include='*_test.go' --color=always -E "errors.Is\([^,]+, nil" . || exit 0
        echo "::error::Replace error.Is nil check with assert.ErrorIs"
        exit 1
```

Something like this (fish) for fixing originals, though this is a first pass:
```
perl -pi -E 'BEGIN{undef $/;}s/if !errors.Is\(err, nil\) {\n[^\n]*\n[^\n]*?}/assert.NoError(t, err)/smg' **/*_test.go
```
Probably do a `git status -s | ... | goimports` to fix any that no longer need errors.

Optional request, obviously.

I'm suggesting this over the less-heinous `if !errors.Is(err, ErrSomeRealThing)`, but maybe we should smash that too.

---

If we are going to add more things another out of scope suggestion:

In sharedtestvalues we have a function `ForceFileStandard` which checks for json tag strings to float64 we could standardise that across exchanges to using `types.Number` and `var EmptyStringPotentialPattern = `.*float64.*json:"[^"]*,string".*`` for the pattern

Then RM that func.

---

My part of this is resolved.
@shazbert - Please Resolve when you're happy too 👍

---

Going to defer this change to another blizzard snow storm 🤣 @thrasher- Will resolve. Thanks @gbjk